x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -432,7 +439,22 @@ public static RoleDescriptor kibanaSystemRoleDescriptor(String name) {
                 // Endpoint metrics. Kibana requires read access to send telemetry
                 RoleDescriptor.IndicesPrivileges.builder()
                     .indices("metrics-endpoint.metrics-*")
-                    .privileges("read").build()
+                    .privileges("read").build(),
+                // Fleet package upgrade
+                RoleDescriptor.IndicesPrivileges.builder()
+                    .indices("logs-*", "synthetics-*", "traces-*",
+                        "/metrics-.*&~(metrics-endpoint\\..*)/")
+                    .privileges("create_index", UpdateSettingsAction.NAME, PutMappingAction.NAME, RolloverAction.NAME)
+                    .build(),
+                // For src/dest indices of the Endpoint package that ships a transform
+                RoleDescriptor.IndicesPrivileges.builder()
+                    .indices("metrics-endpoint.metadata*")
```

Comment:
@joshdover Just to double confirm this is indeed `metrics-endpoint.metadata*`, i.e. without a dash (`-`) right before the wildcard (`*`).